app/controllers/api/logins_controller.rb

```diff
@@ -3,11 +3,19 @@ class Api::LoginsController < Api::ApplicationController
 
   def create
     debate = Debate.find_by code: params[:code]
-    if debate.present?
-      auth_token = debate.auth_tokens.create!
-      render json: { auth_token: auth_token.value }
+    return render json: { error: 'Debate not found' }, status: :not_found unless debate
+    form = MobileUserForm.new(::MobileUser.new)
+    if form.validate(user_params.merge(auth_token: debate.auth_tokens.create!))
+      form.save(validate: false)
```

Comment:
Dziwna logika - najpierw walidujemy i jak się uda, to zapisujemy bez walidacji? Bardziej poprawnie byłoby przypisać parametry (`assign_attributes`), a potem odpalić zwykłego `save`.